### PR TITLE
🏗 CircleCI bash file optimizations

### PR DESCRIPTION
**Pull request by danielrozenberg**
* Move graceful halt logic into separate file
* Store a copy of the merge commit info file in the workspace

Separate step before merging #33708

### CODE REVIEW

File: .circleci/maybe_gracefully_halt.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the license.
+
+# Script used by AMP's CI builds to gracefully halt unnecessary jobs, before
+# executing time-consuming steps such as checking out the code or installing
+# dependencies.
+
+set -e
+
+CYAN() { echo -e "\033[0;36m$1\033[0m"; }
```
Comment (rsimha) [inline .circleci/maybe_gracefully_halt.sh:23]: nit: The convention that has emerged among other shell scripts in `.circleci/` is that successful steps use `GREEN`, errors use `RED`, and special variable names use `CYAN`. So maybe use `GREEN` here?

**Review by rsimha — APPROVED** [review]
One nit below. Quite literally a bike-shed because it has to do with color 😃 

Also a question: Is it your intent to actually invoke this script in a later PR? if so, LGTM. If not, lemme know when I can help with the two-step force-merge dance.

File: .circleci/maybe_gracefully_halt.sh (hunk 1 above)
Comment (danielrozenberg) [reply .circleci/maybe_gracefully_halt.sh]: Done

**Comment by danielrozenberg** [comment]
> … Is it your intent to actually invoke this script in a later PR?

Yes, see https://github.com/ampproject/amphtml/pull/33708#discussion_r610751190

**Comment by danielrozenberg** [comment]
@rsimha added another change here so we can merge multiple config changes into one PR later, PTAL at the new commit

File: .circleci/compute_merge_commit.sh [hunk 2]
```diff
@@ -40,6 +40,11 @@ MERGE_BRANCH="refs/pull/$PR_NUMBER/merge"
 echo $(GREEN "Computing merge SHA of $MERGE_BRANCH...")
 CIRCLE_MERGE_SHA="$(git ls-remote https://github.com/ampproject/amphtml.git "$MERGE_BRANCH" | awk '{print $1}')"
 
+# TODO(danielrozenberg): after #33708 is submitted, remove local directory copy and only store one in the workspace.
```
Comment (rsimha) [inline .circleci/compute_merge_commit.sh:43]: Reminder to update the logging statement on line 49 when this happens.

**Review by rsimha — APPROVED** [review]
👍

File: .circleci/compute_merge_commit.sh (hunk 2 above)
Comment (danielrozenberg) [reply .circleci/compute_merge_commit.sh:43]: Updated it now, might as well... It's technically correct :)

**Pull request merged**